Sources/NIO/Bootstrap.swift

```diff
@@ -66,20 +66,47 @@ public final class ServerBootstrap {
     @usableFromInline
     internal var _childChannelOptions: ChannelOptions.Storage
 
-    /// Create a `ServerBootstrap` for the `EventLoopGroup` `group`.
+    private static func isCompatible(group: EventLoopGroup) -> Bool {
+        return group is SelectableEventLoop || group is MultiThreadedEventLoopGroup
+    }
+
+    /// Create a `ServerBootstrap` on the `EventLoopGroup` `group`.
+    ///
+    /// The `EventLoopGroup` `group` must be compatible, otherwise the program will crash. `ServerBootstrap` is
+    /// compatible only with `MultiThreadedEventLoopGroup` as well as the `EventLoop`s returned by
+    /// `MultiThreadedEventLoopGroup.next`.
```

Comment:
Might be worth highlighting the failable init as another option here and on the other bootstraps.